### PR TITLE
feat(connectors): add Microsoft Teams Connector documentation

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## What is the purpose of the change

add Microsoft Teams Connector documentation

This is draft PR, move to ready for review after https://github.com/camunda/connector-microsoft-teams/issues/3 and https://github.com/camunda/connector-microsoft-teams/issues/4

Link for more comfortable review this docs : https://github.com/camunda/camunda-platform-docs/blob/add_microsoft_teams_documantation/docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md

## Are there related marketing activities

https://github.com/camunda/team-connectors/issues/211

## When should this change go live?

todo

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [ ] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+The **MicrosoftTeams Connector** allows you to connect your BPMN process with [Microsoft Teams](https://www.microsoft.com/microsoft-teams/).
+
+## Prerequisites
+
+To use the **MicrosoftTeams Connector**, you need to have account
+of [Microsoft Teams](https://www.microsoft.com/microsoft-teams/) and
+relevant [permissions](https://support.microsoft.com/en-us/office/manage-team-settings-92d238e6-0ae2-447e-af90-40b1052c4547)
+or registered application in [Azure Active Directory](https://aad.portal.azure.com/) (follow
+guide [how to register app](https://learn.microsoft.com/en-us/graph/auth-register-app-v2) for see more ) with
+relevant [Microsoft Graph API permissions](https://learn.microsoft.com/en-us/graph/permissions-reference).
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:17]:
```suggestion
the relevant [Microsoft Graph API permissions](https://learn.microsoft.com/en-us/graph/permissions-reference).
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+:::note
+All the mandatory and non-mandatory fields depending on the authentication selection you choose are covered in the
+upcoming sections.
+:::
+
+## Authentication
+
+You can choose among the available Microsoft Teams Connectors according to your authentication requirements.
+Microsoft Teams Connector use [Microsoft Graph API](https://learn.microsoft.com/en-us/graph/overview) for work.
+First, you must have a user account in Microsoft Teams with the necessary permissions. See more at
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:43]:
```suggestion
First, you must have a user account with the Microsoft Teams with the necessary permissions. See more at
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+the [Microsoft Teams overview](https://learn.microsoft.com/microsoftteams/teams-overview). If you are not have
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:44]:
```suggestion
the [Microsoft Teams overview](https://learn.microsoft.com/microsoftteams/teams-overview). If you don't have
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+administration role and permissions ask your Microsoft Teams administrator to add needed permissions for work with
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:45]:
```suggestion
administration role and permissions ask your Microsoft Teams administrator to add required permissions for work with
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+Microsoft Teams Connector.
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:46]:
```suggestion
the **Microsoft Teams Connector**.
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+- More information about Microsoft Graph auth you can find here : [Microsoft Graph auth overview](https://learn.microsoft.com/en-us/graph/auth/)
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:48]:
```suggestion
- More information about the Microsoft Graph API authentication you can find here : [Microsoft Graph auth overview](https://learn.microsoft.com/en-us/graph/auth/)
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+### Bearer Token type authentication
+
+For a Token type authentication, take the following steps:
+
+1. Click the **Token** connection type in the **Authentication** section
+2. Set **Token** to `Token`.
+
+![bearer token](../img/connectors-microsoft-teams-bearer-token.png)
+
+See more about [Microsoft Teams Access Token](https://learn.microsoft.com/azure/active-directory/develop/access-tokens);
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:61]:
```suggestion
See more about [Microsoft Teams Access Token](https://learn.microsoft.com/azure/active-directory/develop/access-tokens).
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+#### A few ways for getting Access Token :
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:63]:
```suggestion
#### Options to obtain an access token
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+
+1. Go to [developer.microsoft.com/graph/graph-explorer](https://developer.microsoft.com/graph/graph-explorer);
+   - Login with your Microsoft account;
+   - Choose Tab 'Access Token' and copy Bearer Token :
+     ![connectors-microsoft-teams-copy-token](../img/connectors-microsoft-teams-copy-token.png)
+2. Register your app with the Microsoft identity platform and send a POST request to the '/token' identity platform endpoint to acquire an access token
+   - How to register your App you can read here : [Register your app](https://learn.microsoft.com/en-us/graph/auth-register-app-v2)
+   - How to get access on behalf of a user you can read here : [Get access on behalf of a user](https://learn.microsoft.com/en-us/graph/auth-v2-user)
+   - How to get access without a user you can read here : [Get access without a user](https://learn.microsoft.com/en-us/graph/auth-v2-service)
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:72]:
```suggestion
1. Via the graph explorer
   - Visit [developer.microsoft.com/graph/graph-explorer](https://developer.microsoft.com/graph/graph-explorer)
   - Login with your Microsoft account
   - Choose tab 'Access Token' and copy the bearer token
   
![connectors-microsoft-teams-copy-token](../img/connectors-microsoft-teams-copy-token.png)

2. Register your app with the Microsoft identity platform and send a POST request to the `/token` identity platform endpoint to acquire an access token
   - How to register your App: [Register your app](https://learn.microsoft.com/en-us/graph/auth-register-app-v2)
   - How to get access on behalf of a user: [Get access on behalf of a user](https://learn.microsoft.com/en-us/graph/auth-v2-user)
   - How to get access without a user: [Get access without a user](https://learn.microsoft.com/en-us/graph/auth-v2-service)
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+
+### Client credentials type authentication
+
+For a Client credentials type authentication, take the following steps:
+
+1. Click the **Client credentials** connection type in the **Authentication** section
+2. Set **Tenant id** to `Tenant ID`. (Your Microsoft Teams tenant ID is a unique identifier. See more [How to find Tenant ID](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-how-to-find-tenant))
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:79]:
```suggestion
2. Set **Tenant id** to `Tenant ID`. (Your Microsoft Teams tenant ID is a unique identifier. See more about [how to find a tenant ID](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-how-to-find-tenant))
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+3. Set **Client id** to `Client ID`. (The application ID that the [Azure app registration portal](https://go.microsoft.com/fwlink/?linkid=2083908) assigned to your app.)
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:80]:
```suggestion
3. Set **Client id** to `Client ID`. (The application ID that the [Azure app registration portal](https://go.microsoft.com/fwlink/?linkid=2083908) assigned to your app)
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+4. Set **Secret id** to `Secret ID`. (The client secret that you created in the app registration portal for your app)
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:81]:
```suggestion
4. Set the **Secret ID** field: the client secret that you created in the app registration portal for your app
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md (hunk 11 above)
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:80]:
```suggestion
3. Set the **Client ID** field: the application ID that the [Azure app registration portal](https://go.microsoft.com/fwlink/?linkid=2083908) assigned to your app
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+
+![client credentials](../img/connectors-microsoft-teams-client-credentials.png)
+
+## Conversation type and method
+
+In the **Operation** section you could choose conversation type, for now it is **Chat** and **Channel**, after that you need choose one of suggestion method :
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:87]:
```suggestion
In the **Operation** section you could choose a conversation type, either **Chat**, or **Channel**, after that you need to choose one of suggested method:
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+
+![todo add gif](../img/connectors-microsoft-teams-choose-method.gif)
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:89]:
```suggestion
![MS Teams Choose Method](../img/connectors-microsoft-teams-choose-method.gif)
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+
+:::note
+For example, if you want to send message in Microsoft Teams channel, you need choose conversation type **Channel** and method **Send message in channel**
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:92]:
```suggestion
For example, if you want to send a message in Microsoft Teams channel, you need choose conversation type **Channel** and method **Send message in channel**
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+:::
+
+## Data section
+
+### Chat conversation type
+
+#### Properties:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:99]:
```suggestion
#### Properties
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+
+Members property must contain list of Member:
+
+|     Property      |     Type     | Required                               |
+| :---------------: | :----------: | -------------------------------------- |
+|      userId       |    string    | Yes, if 'userPrincipalName' is not set |
+| userPrincipalName |    string    | Yes, if 'userId' is not set            |
+|       roles       | string array | Yes                                    |
+
+Example :
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:123]:
```suggestion
Example:
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+    "roles": ["owner"]
+  },
+  {
+    "principalName": "john.dou@mail.com",
+    "roles": ["owner"]
+  }
+]
+```
+
+#### Methods:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:140]:
```suggestion
#### Methods
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 19]
```diff
@@ -0,0 +1,253 @@
...
+|  Create a new chat   |          [https://learn.microsoft.com/en-us/graph/api/chat-post](https://learn.microsoft.com/en-us/graph/api/chat-post?view=graph-rest-1.0&tabs=http)          |
+|    Get chat by ID    |           [https://learn.microsoft.com/en-us/graph/api/chat-get](https://learn.microsoft.com/en-us/graph/api/chat-get?view=graph-rest-1.0&tabs=http)           |
+|  List chat members   |  [https://learn.microsoft.com/en-us/graph/api/chat-list-members](https://learn.microsoft.com/en-us/graph/api/chat-list-members?view=graph-rest-1.0&tabs=http)  |
+| Send message in chat | [https://learn.microsoft.com/en-us/graph/api/chat-post-messages](https://learn.microsoft.com/en-us/graph/api/chat-post-messages?view=graph-rest-1.0&tabs=http) |
+| Get message in chat  |    [https://learn.microsoft.com/en-us/graph/api/chatmessage-get](https://learn.microsoft.com/en-us/graph/api/chatmessage-get?view=graph-rest-1.0&tabs=http)    |
+|  List chat members   |  [https://learn.microsoft.com/en-us/graph/api/chat-list-members](https://learn.microsoft.com/en-us/graph/api/chat-list-members?view=graph-rest-1.0&tabs=http)  |
+
+### Channel conversation type
+
+#### Properties:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:153]:
```suggestion
#### Properties
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 20]
```diff
@@ -0,0 +1,253 @@
...
+|           Top           |                                                                                 List channel messages                                                                                  |    No    | numbers  |                                       Controls the number of items per response                                        |
+|      With replies       |                                                                                 List channel messages                                                                                  |   Yes    | boolean  |          Choose 'FALSE' for get messages without replies<br/>Choose 'FALSE' for get messages without replies           |
+|       Message ID        |                                                                                  List message replies                                                                                  |   Yes    |  string  |                                           Microsoft Teams channel message id                                           |
+
+#### Methods:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:170]:
```suggestion
#### Methods
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 21]
```diff
@@ -0,0 +1,253 @@
...
+| Send message to channel | [https://learn.microsoft.com/en-us/graph/api/channel-post-messages](https://learn.microsoft.com/en-us/graph/api/channel-post-messages?view=graph-rest-1.0&tabs=http)       |
+|   Get channel message   | [https://learn.microsoft.com/en-us/graph/api/chatmessage-get](https://learn.microsoft.com/en-us/graph/api/chatmessage-get?view=graph-rest-1.0&tabs=http)                   |
+|  List channel messages  | [https://learn.microsoft.com/en-us/graph/api/channel-list-messages](https://learn.microsoft.com/en-us/graph/api/channel-list-messages?view=graph-rest-1.0&tabs=http)       |
+|  List message replies   | [https://learn.microsoft.com/en-us/graph/api/chatmessage-list-replies](https://learn.microsoft.com/en-us/graph/api/chatmessage-list-replies?view=graph-rest-1.0&tabs=http) |
+|      List members       | [https://learn.microsoft.com/en-us/graph/api/channel-list-members](https://learn.microsoft.com/en-us/graph/api/channel-list-members?view=graph-rest-1.0&tabs=http)         |
+
+## Microsoft Teams Connector response
+
+The **Microsoft Teams Connector** returns the Microsoft Graph API response in 'result' wrapper:
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:185]:
```suggestion
The **Microsoft Teams Connector** returns the Microsoft Graph API response in `result` wrapper:
```

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 22]
```diff
@@ -0,0 +1,253 @@
...
+      }
+    },
+    "tenantId": "0000000-0000-0000-0000-000000000",
+    "webUrl": "https://teams.microsoft.com/l/chat/19%3Aefb08ac3-0000f-0000-0000-example-chat-id_fe35bf61-0000-0000-0000-ddc97d8903d4%40unq.gbl.spaces/0?tenantId=00000-0000-0000-0000-00000000",
+    "id": "19%3Aefb08ac3-0000f-0000-0000-example-chat-id_fe35bf61-0000-0000-0000-ddc97d8903d4%40unq.gbl.spaces"
+  }
+}
+```
+
+See [channel resource type](https://learn.microsoft.com/graph/api/resources/channel?view=graph-rest-1.0) for find response on needed method for channel conversation type or see [chat resource type](https://learn.microsoft.com/graph/api/resources/chat?view=graph-rest-1.0) for find response on needed method for chat conversation type.
```
Comment (igpetrov) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:234]:
```suggestion
See [channel resource type](https://learn.microsoft.com/graph/api/resources/channel?view=graph-rest-1.0) to find response for the required method for a channel conversation type or see [chat resource type](https://learn.microsoft.com/graph/api/resources/chat?view=graph-rest-1.0) to find response for the required method for a chat conversation type.
```

**Review by igpetrov — CHANGES_REQUESTED** [review]
What about limitations? AFAIK, you can only send messages with a bearer token but not with client credentials, or did you resolve that?

**Comment by igpetrov** [comment]
@Oleksiivanov is this ready for review?

**Comment by igpetrov** [comment]
@Oleksiivanov please keep in mind that the build is failing

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 23]
```diff
@@ -0,0 +1,248 @@
...
+|        Group ID         | Create channel <br/> Get channel <br/> List channels <br/> Send message to channel <br/> Get channel message <br/> List channel messages <br/> List message replies <br/> List members |   Yes    |  string  |                                                Microsoft Teams group id                                                |
+|       Channel ID        |           Get channel <br/> List channels <br/> Send message to channel <br/> Get channel message <br/> List channel messages <br/> List message replies <br/> List members            |   Yes    |  string  |                                               Microsoft Teams channel id                                               |
+|      Display name       |                                                                                     Create channel                                                                                     |    No    |  string  |                                         Displayed name of new MS teams channel                                         |
```
Comment (christinaausley) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:154]: When you say "MS teams channel" do you mean Microsoft Teams channel? If so, I would spell this out formally.

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 24]
```diff
@@ -0,0 +1,248 @@
...
+|       Description       |                                                                                     Create channel                                                                                     |    No    |   text   |                                          Description of new MS teams channel                                           |
+| Channel membership type |                                                                                     Create channel                                                                                     |   Yes    | dropdown | See [teams-channels-overview](https://learn.microsoft.com/microsoftteams/teams-channels-overview) for more information |
+|          Owner          |                                                                Create channel (if Channel membership type != STANDARD)                                                                 |   Yes    |  string  |                               Channel owner; MS teams user id or MS teams principal name                               |
```
Comment (christinaausley) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:157]: Same comment here.

File: docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md [hunk 25]
```diff
@@ -0,0 +1,248 @@
...
+2. Use **Result Expression** to map fields from the response into process variables. For example:
+
+```
+= {
+  "chatId": result.id,
+  "tenantId": result.tenantId
+}
+```
+
+## Appendix & FAQ
```
Comment (christinaausley) [inline docs/components/connectors/out-of-the-box-connectors/microsoft-teams.md:243]: As this is mentioned on line 18 and users are directed here, and then again directed to managing secrets, would it be better to remove this appendix section and on line 18 just say something like:
:::note
It is highly recommended to use Camunda secrets to store credentials so you don't expose sensitive information directly from the process. See [managing secrets](../../../components/console/manage-clusters/manage-secrets.md) to learn more.
:::

**Review by christinaausley — CHANGES_REQUESTED** [review]
Technical review in latest commit for wording, grammar, punctuation, and more. Let me know if you have any questions about these changes.

Additionally, I had a few comments regarding an abbreviation and the FAQ section.

**Comment by pepopowitz** [comment]
Hi! This PR is currently in a "frozen" state, with merging blocked due to [a new `enforce-label` check](https://github.com/camunda/camunda-platform-docs/pull/1578) not having run.

This is fixable! If you rebase on `main`, or merge `main`, this branch will get the changes it needs to run that check. This PR has a `hold` label, so when that check runs it will put the PR in a _different_ un-mergeable state -- but at least that one will be correct! 

If you'd prefer, @akeller or myself have the ability to check a box to merge without the check running. If/when you've got the reviews you need and you want us to merge for you, just let us know.

**Review by christinaausley — APPROVED** [review]
Thanks for resolving my comments! 👍

**Review by igpetrov — APPROVED** [review]
LGTM

**Pull request merged**